extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {user} from '../../../src/log';
+
+
+/** @const */
+// const EXPERIMENT = 'amp-story-auto-ad';
+
+/** @const */
+const TAG = 'amp-story-auto-ad';
+
+export class AmpStoryAutoAds extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+    this.ampStory = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    const ampStory = this.element.parentElement;
+    user().assert(ampStory.tagName === 'AMP-STORY',
+        `<${TAG}> should be child of <amp-story>`);
+    ampStory.appendChild(this.makeMockPage());
```

Comment:
we might want to do this in layoutCallback.

Reply:
my thought was that we wanted to let the ad resources start loading ASAP but if you want `layoutCallback` SGTM

Reply:
we want to be nice to let content render first.
we are not showing ads in the first 3 page views in anyway.